shotover-proxy/src/transforms/protect/mod.rs

```diff
@@ -227,33 +228,38 @@ impl Transform for Protect {
                                             .protect(&self.key_source, &self.key_id)
                                             .await?;
                                         **value = SQLValue::from(&MessageValue::from(protected));
+                                        invalidate_cache = true;
                                     }
                                 }
                             }
                         }
                     }
                 }
             }
+            if invalidate_cache {
+                message.invalidate_cache();
+            }
         }
 
         let mut original_messages = message_wrapper.messages.clone();
         let mut result = message_wrapper.call_next_transform().await?;
 
         for (response, request) in result.iter_mut().zip(original_messages.iter_mut()) {
-            if let Frame::Cassandra(CassandraFrame {
+            let mut invalidate_cache = false;
+            if let Some(Frame::Cassandra(CassandraFrame {
                 operation:
                     CassandraOperation::Result(CassandraResult::Rows {
                         value: MessageValue::Rows(rows),
                         ..
                     }),
                 ..
-            }) = &mut response.original
+            })) = response.frame()
             {
                 if let Some(namespace) = request.namespace() {
-                    if let Frame::Cassandra(CassandraFrame {
+                    if let Some(Frame::Cassandra(CassandraFrame {
                         operation: CassandraOperation::Query { query, .. },
                         ..
-                    }) = &mut request.original
+                    })) = &mut request.frame()
```

Comment:
Do we need this `&mut`?

Reply:
Thanks, cleaned up a whole bunch of those